internal/server/services/backup_schedule/backup_schedule_service.go

```diff
@@ -332,13 +338,76 @@ func (s *BackupScheduleService) ListBackupSchedules(
 }
 
 func (s *BackupScheduleService) ToggleBackupSchedule(
-	ctx context.Context, in *pb.ToggleBackupScheduleRequest,
+	ctx context.Context, request *pb.ToggleBackupScheduleRequest,
 ) (*pb.BackupSchedule, error) {
 	ctx = grpcinfo.WithGRPCInfo(ctx)
-	ctx = xlog.With(ctx, zap.String("BackupScheduleID", in.GetId()))
-	xlog.Error(ctx, "ToggleBackupSchedule not implemented")
-	//TODO implement me
-	return nil, status.Error(codes.Internal, "not implemented")
+
+	scheduleID := request.GetId()
+	ctx = xlog.With(ctx, zap.String("BackupScheduleID", scheduleID))
+
+	xlog.Debug(ctx, "ToggleBackupSchedule", zap.Stringer("request", request))
+
+	schedules, err := s.driver.SelectBackupSchedulesWithRPOInfo(
+		ctx, queries.NewReadTableQuery(
+			queries.WithRawQuery(GetScheduleQuery),
```

Comment:
I think you do need to join schedules with backups for this request, and we can just select from Schedules